### DISCUSSION TITLE
How to calculate total energy consumptions used by all datacenter

### DISCUSSION THREAD

**Discussion by LeeBoonKong** [manoelcampos/cloudsim-plus#355] [Q&A] [answered]
Hi all, 
I am new to Cloudsim Simulations and I am currently doing a project that requires me to calculate the total energy used by all datacenters after executing all cloudlets.
How can I obtain the total energy consumption?

Thank You

**Comment by manoelcampos** [comment, marked as answer]
Check [this example](https://github.com/manoelcampos/cloudsimplus/blob/master/cloudsim-plus-examples/src/main/java/org/cloudsimplus/examples/power/PowerExample.java). You just need to sum up power consumption for each host in each datacenter.

**Reply by LeeBoonKong** [reply]
Ahh, thanks a lot for your reply! I'll go check it out

**Comment by LeeBoonKong** [comment]


**Reply by manoelcampos** [reply]
Could you please open a new question?

**Reply by TaliaAsser** [reply]
I want to apply PSO in cloudsim to optimize vm placement _

**Comment by jlxcsong** [comment]
您好！您发的邮件我已收到！
